### ISSUE TITLE
Add ARM64 support to Rundeck

### ISSUE THREAD

**Issue by odidev** [rundeck/rundeck#6433]
Hi Team,

I am looking to add ARM64 jobs in Travis.
I have checked that the code is currently structured to support the AMD64 platform, hence Travis jobs for ARM64 are failing. Also, it seems that adding ARM64 support to Rundeck would need a good amount of changes in the code.

It will be really helpful if ARM64 support is added to the package. Do you have any plans for the same?

If you are interested, I can add ARM64 support to Rundeck and raise PR.

Also, it will be really helpful if you can share any blockers for this activity.

Please provide your opinion on this.

**Comment by fdevans** [comment]
There are currently not active plans to support Rundeck on ARM64.  As we start to see more requests we'll review the effort to add this.

**Comment by odidev** [comment]
@fdevans , thanks for your response. Please let me know whether you are interested in accepting any PR for adding rundeck support on arm64. I am happy to help on making rundeck available for aarch64.

**Comment by odidev** [comment]
@fdevans I was trying to run travis build for arm64 but it was failing with below error:
standard_init_linux.go:211: exec user process caused "exec format error"

This package uses rundeckapp/testdeck:base-latest image which is not available for arm64. I have found the [github](https://github.com/rundeck/testdeck) link for this image, but I am unable to find its Dockerfile in it to add support for arm64.

Could you please point me to Dockerfile source, so that I can add arm64 support in it and raise PR?

**Comment by foxhunt72** [comment]
Please i would second this request

**Comment by DarwisNarvaezDev** [comment]
Hi, im currently having issues to set docker images of RD on my m1 mac. So i add my request too!

**Comment by fdevans** [comment]
@odidev Yes, we do review community PRs.

**Comment by agarwalpranab** [comment]
Would love this. +1

**Comment by matttobes** [comment]
I would like to see this

**Comment by calebcheng** [comment]
Got Apple Macbook Pro with M1 chip and really need this. I believe there will be many other people having the same issue

**Comment by augustgerro** [comment]
I would like to see M silicon as well

**Comment by florian-asche** [comment]
There is an existing java war file. That should work on arm64 or am i wrong?

**Comment by c-geek** [comment]
I've successfully built Rundeck for ARM (here on an M1 Mac), I won't make a PR but you can build it too using this branch:

    git clone https://github.com/c-geek/rundeck.git -b fix/6433/arm
    cd rundeck
    ./gradlew :docker:officialBuild

And then run it:

    docker run --name some-rundeck -p 4440:4440 rundeck/rundeck:SNAPSHOT

![image](https://user-images.githubusercontent.com/969136/209401695-24cedace-64a2-4e6d-ad2e-7cf2bda21e50.png)

**Comment by georgedimac** [comment]
@c-geek , 
context: macos with jdk 19

first i need to install _git-lfs_ because i was getting an 
>git-lfs filter-process: git-lfs: command not found
>fatal: the remote end hung up unexpectedly
>warning: Clone succeeded, but checkout failed.


1. getting 'Unsupported class file major version 63'
      i had to change to _gradle 7.6_ (the latest one and the one that supports jdk 19) in _gradle-wrapper.properties_

2. next error: 

> An exception occurred applying plugin request [id: 'org.dvaske.gradle.git-build-info', version: '0.8']
> Failed to apply plugin 'org.dvaske.gradle.git-build-info'.
> No signature of method: org.gradle.internal.logging.slf4j.OutputEventListenerBackedLogger.warning() is applicable for > argument types: (org.codehaus.groovy.runtime.GStringImpl) values: [Git repository not found for rundeck]

ok , changed it to version 0.9 in _build.gradle_

4. the hell of incompatibility occured.

<img width="1189" alt="Screenshot 2022-12-24 at 10 51 04" src="https://user-images.githubusercontent.com/112869370/209428364-687c5790-921b-4f4e-aeaa-9ba19ad4f8bf.png">

-----

so it seems it is hard to make it compatible with the latest jdk, as i read on the documentation it needs 8 or 11. 

how can i specify to select them , they are also installed :

>% java -version
>java version "19" 2022-09-20
>Java(TM) SE Runtime Environment (build 19+36-2238)
>Java HotSpot(TM) 64-Bit Server VM (build 19+36-2238, mixed mode, sharing)
>
>% /usr/libexec/java_home -V
>Matching Java Virtual Machines (4):
>    19 (arm64) "Oracle Corporation" - "Java SE 19" /Library/Java/JavaVirtualMachines/jdk-19.jdk/Contents/Home
>    11.0.16.1 (x86_64) "Microsoft Build of OpenJDK" - "Microsoft Build of OpenJDK 11" /Library/Java/JavaVirtualMachines/microsoft-11.jdk/Contents/Home
>    1.8.351.10 (x86_64) "Oracle Corporation" - "Java" /Library/Internet Plug-Ins/JavaAppletPlugin.plugin/Contents/Home
>    1.8.0_302 (x86_64) "Eclipse Temurin" - "Eclipse Temurin 8" /Library/Java/JavaVirtualMachines/temurin-8.jdk/Contents/Home

**Comment by zuoyebushiwo** [comment]
Would love this. +1

**Comment by emmiga** [comment]
+1

**Comment by yaroze** [comment]
I agree this should be multiplatform, especially when there are people migrating to ARM64 in cloud environments.
Any news on officially supporting this? It seems @c-geek has already done a great work!

**Comment by jieey1140** [comment]
I hope this issue will be resolved soon. m1 MacBook turns on super slow and is so slow that I can't use it :(

**Comment by c-geek** [comment]
@georgedimac Rundeck 4.9.0 needs Java 1.8 to compile. On my Mac M1:

````
git clone https://github.com/c-geek/rundeck.git -b fix/6433/arm
cd rundeck
export JAVA_HOME=`/usr/libexec/java_home -v 1.8.0_352`
./gradlew :docker:officialBuild
```

**Comment by ChuckCrawford** [comment]
We are not publishing ARM64 images at this time but https://github.com/rundeck/rundeck/pull/9141 should enable folks to successfully build ARM64 images on their own machines.

**Comment by fdevans** [comment]
Thanks for the report. As part of a cleanup of our issue tracker, we're closing issues older than 12 months, in line with Rundeck's support policy of one year from a release's date (see the [release calendar](https://docs.rundeck.com/docs/history/release-calendar.html) for currently supported versions). This isn't a judgment that the report is invalid — many of these are likely still real — but we can't verify against a version this old.

If you're still seeing this on a currently supported release (6.1.x or 5.20.x+), please open a new issue using our [bug report template](https://github.com/rundeck/rundeck/issues/new?template=bug_report.yml) with your version and reproduction steps. Please don't reopen this issue — a fresh issue keeps things tracked against a supported version and gets it in front of us properly. We're actively working on improving how issues get submitted and triaged going forward.

Thanks for your patience,
The Rundeck Team
